spec/unit/transaction_spec.rb

```diff
@@ -420,6 +420,18 @@ def generate
       transaction.evaluate
     end
 
+    it "should call ::pre_resource_eval on provider classes that support it" do
+      selinux = double('selinux', is_selinux_enabled: true)
```

Comment:
I think something like this will work:

```suggestion
      selinux = double('selinux', selinux_open: handle)
      allow(Puppet::Util::SELinux).to receive(:is_selinux_enabled?).and_return(true)
```

You'll need to define `handle` in a let block if it's not already.